### DISCUSSION TITLE
Reading a real tag - possible issue?

### DISCUSSION THREAD

**Discussion by sdmolt** [ottowayi/pycomm3#121] [Q&A] [answered]
Couldn't find anything on this one - When reading a REAL tag, I get a number back that is slightly off - for example:

1. real_tag in PLC = -0.1, pycomm3 reads: `Tag(tag='real_tag' value=-0.10000000149011612, type='REAL', error=None)`
2. real_tag in PLC = -0.01, pycomm3 reads: `Tag(tag='real_tag', value=0.009999999776482582, type='REAL', error=None)`

I also noticed the responses in the docs (basic read/write examples), your response for the real was `Tag(tag='REAL1', value=100.0009994506836, type='REAL', error=None)` - was that the actual value in the processor?

**Comment by ottowayi** [comment, marked as answer]
Yeah that is because of how floating points are represented in binary.  [Here](https://docs.python.org/3/tutorial/floatingpoint.html) is a doc that explains more about how Python handles floats.  The plc is doing some rounding to show you the closest approximate value and python does the same when you type `.1` or `1/10`.  But, since I'm using the struct module to unpack the floating point value from bytes,  I don't think it will do the rounding for the string representation.  One thing I could do is change the repr for the Tag objects to round to a certain number of decimal places, but then it would be to a fixed number of decimals for every object.

**Reply by sdmolt** [reply]
I think the way it is now is probably good rather than defining a decimal count. I can do the rounding outside of pycomm3. Thanks for the fast response!
